src/Cli/dotnet/commands/dotnet-workload/search/WorkloadSearchVersionsCommand.cs

```diff
@@ -0,0 +1,105 @@
+
+            _sdkVersion = creationResult.SdkVersion;
+            var workloadResolver = creationResult.WorkloadResolver;
+
+            _numberOfWorkloadSetsToTake = result.GetValue(WorkloadSearchVersionsCommandParser.TakeOption);
+            _workloadSetOutputFormat = result.GetValue(WorkloadSearchVersionsCommandParser.FormatOption);
+
+            // For these operations, we don't have to respect 'msi' because they're equivalent between the two workload
+            // install types, and FileBased is much easier to work with.
+            _installer = new FileBasedInstaller(
+                reporter,
+                new SdkFeatureBand(_sdkVersion),
+                workloadResolver,
+                CliFolderPathCalculator.DotnetUserProfileFolderPath,
+                nugetPackageDownloader: null,
+                dotnetDir: Path.GetDirectoryName(Environment.ProcessPath),
+                tempDirPath: null,
+                verbosity: Verbosity,
+                packageSourceLocation: null,
+                restoreActionConfig: new RestoreActionConfig(result.HasOption(SharedOptions.InteractiveOption)),
+                nugetPackageDownloaderVerbosity: VerbosityOptions.quiet
+                );
+
+            _workloadVersion = result.GetValue(WorkloadSearchVersionsCommandParser.WorkloadVersionArgument);
+        }
+
+        public override int Execute()
+        {
+            if (_workloadVersion is null)
+            {
+                var featureBand = new SdkFeatureBand(_sdkVersion);
+                var packageId = _installer.GetManifestPackageId(new ManifestId("Microsoft.NET.Workloads"), featureBand);
+                var versions = PackageDownloader.GetLatestPackageVersions(packageId, _numberOfWorkloadSetsToTake, packageSourceLocation: null, includePreview: !string.IsNullOrWhiteSpace(_sdkVersion.Prerelease))
+                    .GetAwaiter().GetResult()
+                    .Select(version => WorkloadManifestUpdater.WorkloadSetPackageVersionToWorkloadSetVersion(featureBand, version.Version.ToString()));
+                if (_workloadSetOutputFormat?.Equals("json", StringComparison.OrdinalIgnoreCase) == true)
+                {
+                    Reporter.WriteLine(JsonSerializer.Serialize(versions.Select(version => version.ToDictionary(_ => "workloadVersion", v => v))));
+                }
+                else
+                {
+                    Reporter.WriteLine(string.Join('\n', versions));
+                }
+            }
+            else
+            {
+                var workloadSet = _installer.GetWorkloadSetContents(_workloadVersion);
+                if (_workloadSetOutputFormat?.Equals("json", StringComparison.OrdinalIgnoreCase) == true)
+                {
+                    Reporter.WriteLine(JsonSerializer.Serialize(
+                        workloadSet.ManifestVersions.ToDictionary(kvp => kvp.Key.ToString(), kvp => $"{kvp.Value.Version}/{kvp.Value.FeatureBand}")));
```

Comment:
I think this is (and should be) the same format as the rollback file.  So use `WorkloadSet.ToJson` here instead of duplicating the logic of how it's formatted.